Sources/XCTest/XCTestCase.swift

```diff
@@ -26,7 +26,11 @@ public typealias XCTestCaseEntry = (testCaseClass: XCTestCase.Type, allTests: [(
 
 public class XCTestCase {
 
-    public required init() {
+    /// The name of the test case, consisting of its class name and the method name it will run
+    public let name: String
+
+    public required init(methodName: String) {
```

Comment:
Thanks for pointing this out. When I first saw your comment I was pretty sure I had simply made that constructor `public` by mistake. Attempting to change it landed me in the middle of an intense battle with the compiler, which ended with me conceding we will have to live with `name` having a public setter for now (I left comments in the new code pointing this out.) Sadly non-public fields seem to be off-limits for now :disappointed:

I'm reasonably content with the latest version of this, though. What do you think?

(As an aside, while trying to understand the bigger picture here, I learned that it's effectively impossible to provide an overridden constructor on Swift `XCTestCase` subclasses under Apple XCTest right now, because the framework calls `init(invocation:)` when building the default test suite, which can't be overridden because of `NSInvocation`'s unavailability.)